### PR TITLE
Multiprocessing for structural modeling

### PR DESCRIPTION
**Pull request by schallerdavid**
## Description
Add support for multiprocessing for KLIFS complex featurizers. This PR will also alter abstract featurizer classes in `kinoml/features/core` and subclasses thereof.

## Todos
  - [x] add `_pre_featurize` method to `BaseFeaturizer`
  - [x] move KLIFS queries to `_pre_featurize` method of KLIFS complex featurizers
  - [x] KLIFS complex featurizers return empty MDAnalysis universes in case no valid kinase structure is available to model the given system
  - [x] move multiprocessing arguments to `__init__` of BaseFeaturizer
    - [x] default use_multiprocessing=False
    - [x] if use_multiprocessing=True, use all available threads
    - [x] n_processes argument to select number of threads
  - [x] adapt all classes in `kinoml/features/core` to new multiprocessing handling
  - [x] add arguments for custom cache and output directory
  - [x] allow keeping the protonation state of the molecule to dock
  - [x] implement multiprocessing handling via dask
 
## Other improvements
  - [x] add support to select specific PDB structures for modeling in OEKLIFSKinaseHybridDockingFeaturizer

## Status
- [ ] Ready to go

### CODE REVIEW

**Comment by schallerdavid** [comment]
Hi @t-kimber ,
everything seems to work out with the different parallelization schemes. I also introduced a few changes to ligand featurizers because of the changes I made to the basefeaturizers. Do you wanna give it a try or have any suggestions before I merge?

**Review by jchodera — APPROVED** [review]
We reviewed this PR collectively, and we're good to merge once tests pass!

**Review by t-kimber — APPROVED** [review]

**Pull request merged**
